activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
I agree with @egilburg that we shouldn't treat a new object differently based on whether all its companions are new, which this conditional is clearly doing. If we're *currently* doing that, and this instead means everything ends up lining up, we should be finding & fixing the existing conditional to be more generous... frankly, I'm suspicious about a bug fix with this much green in the diff.